metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -450,8 +452,48 @@ def _find_measure_recipe(
                 for y in x.join_on_partition_time_dimensions
             )
 
+            measure_node = node_with_lowest_cost
+            # Since every build_aggregate_measures function call groups measure_specs by additiveness
+            # all of these measure_specs will have the same non_additive_dimension_spec
+            non_additive_dimension_spec: Optional[NonAdditiveDimensionSpec] = measure_specs[
+                0
+            ].non_additive_dimension_spec
+            if non_additive_dimension_spec is not None:
+                # Gets the dimension given in the query that matches the non-additive dimension
+                primary_time_dimension = self._data_source_semantics.find_primary_time_dimension(
+                    self._data_source_semantics.get_data_sources_for_measure(measure_specs[0].as_reference)[0]
+                )
```

Comment:
These two accessors are making assumptions about how this method will be called, and they'll produce incorrect results if those assumptions ever change.

I'd prefer to make this part of the method signature, or find some other way of ensuring that this is correct. 

We could also bundle these up top, where the assertion about the data sources lives, and add invariant checks at that point.

Reply:
addressed here https://github.com/transform-data/metricflow/pull/253#discussion_r977898952